packages/components/src/presentation-components/properties/UniquePropertyValuesSelector.tsx

```diff
@@ -143,12 +131,16 @@ function getBaseClassInfo(field?: Field) {
   if (field?.parent === undefined && field?.isPropertiesField()) {
     return field.properties[0].property.classInfo;
   }
-  const lastStepToPrimaryClass = field?.parent?.pathToPrimaryClass.slice(-1).pop();
+
+  let deepestParentField = field?.parent;
```

Comment:
Variable naming should be switched around to `rootParentField` as you are starting from nested field and climbing up.